packages/thrift-server-core/src/main/protocols/BinaryProtocol.ts

```diff
@@ -138,12 +140,14 @@ export class BinaryProtocol extends TProtocol {
         }
     }
 
-    public writeI64(i64: number | Int64): void {
+    public writeI64(i64: number | string | IInt64): void {
```

Comment:
TProtocol defined:
```
public abstract writeI64(i64: number | Int64): void
```

However, multiple implementations checked for `string` types.

The function now consistently accepts `number | string | IInt64` across protocols.